langchain/chains/combine_documents/map_reduce.py

```diff
@@ -21,6 +24,9 @@ class MapReduceDocumentsChain(BaseCombineDocumentsChain, BaseModel):
     document_variable_name: str
     """The variable name in the llm_chain to put the documents in.
     If only one variable in the llm_chain, this need not be provided."""
+    question_prompt: PromptTemplate
```

Comment:
what are these used for? they should actually be part of llm_chain and combine_document_chain

Reply:
It's just `document_prompt` we need for the computing of the num of tokens in a full prompt of all the docs pulled together.

Reply:
This commit https://github.com/hwchase17/langchain/pull/338/commits/1a447ce272022a7b778be1a8f145ec40c0c558ed removes the other two prompts that are not needed.